packages/select/src/utils/getLGSelectTestUtils/getLGSelectTestUtils.spec.tsx

```diff
@@ -0,0 +1,466 @@
+
+import { Option, OptionGroup, Select, SelectProps, State } from '../../';
+
+import { getLGSelectTestUtils } from './getLGSelectTestUtils';
+
+const defaultProps = {
+  label: 'Label',
+  name: 'pet',
+  description: 'Description',
+} as const;
+
+const children = [
+  <Option key="red" value="RED">
+    Red
+  </Option>,
+  <Option key="blue">Blue</Option>,
+  <Option key="orange" disabled>
+    Orange you glad
+  </Option>,
+  <OptionGroup key="enabled group" label="Enabled group">
+    <Option>Green</Option>
+    <Option>Yellow</Option>
+  </OptionGroup>,
+  <OptionGroup key="disabled group" label="Disabled group" disabled>
+    <Option>Indigo</Option>
+    <>
+      <Option>Violet</Option>
+    </>
+  </OptionGroup>,
+];
+
+function renderSelect(props = {}) {
+  const renderUtils = render(
+    <Select {...defaultProps} {...props}>
+      {children}
+    </Select>,
+  );
+
+  const { elements, utils } = getLGSelectTestUtils();
+
+  return {
+    ...renderUtils,
+    ...elements,
+    ...utils,
+  };
+}
+
+function renderSelectControlled(props = {}) {
+  const ControlledSelect = (props = {}) => {
+    // @ts-expect-error
+    // eslint-disable-next-line react/prop-types
+    const { value: propValue, ...rest } = props;
+    const [value, setValue] = useState(propValue);
+
+    return (
+      <Select {...defaultProps} {...rest} value={value} onChange={setValue}>
+        {children}
+      </Select>
+    );
+  };
+
+  const renderUtils = render(<ControlledSelect {...props} />);
+  const { elements, utils } = getLGSelectTestUtils();
+
+  return {
+    ...renderUtils,
+    ...elements,
+    ...utils,
+  };
+}
+
+function renderMultipleSelects() {
+  const renderUtils = render(
+    <>
+      <Select {...defaultProps} defaultValue="RED" data-lgid="lg-select-1">
+        {children}
+      </Select>
+      <Select {...defaultProps} defaultValue="Blue" data-lgid="lg-select-2">
+        {children.slice(0, 2)}
+      </Select>
+    </>,
+  );
+
+  const { elements: testElements1, utils: testUtils1 } =
+    getLGSelectTestUtils('lg-select-1');
+  const { elements: testElements2, utils: testUtils2 } =
+    getLGSelectTestUtils('lg-select-2');
+
+  return {
+    ...renderUtils,
+    testElements1,
+    testElements2,
+    testUtils1,
+    testUtils2,
+  };
+}
+
+describe('packages/select/getLGSelectTestUtils', () => {
+  test('throws error if select is not found', () => {
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { getLabel } = renderSelect({ 'data-lgid': 'different-id' });
+    } catch (error) {
+      expect(error).toBeInstanceOf(Error);
+      expect(error).toHaveProperty(
+        'message',
+        expect.stringMatching(
+          /Unable to find an element by: \[data-lgid="lg-select"\]/,
+        ),
+      );
+    }
+  });
+
+  describe('getLabel', () => {
+    test('renders label', () => {
+      const { getLabel } = renderSelect();
+      expect(getLabel()).toBeInTheDocument();
```

Comment:
- I know it's more verbose, but since this suite is testing the `getLGSelectTestUtils` function, we should be calling that function in each test instead of abstracting it into a `renderSelect` function